qutip/solver/flimesolve.py

```diff
@@ -0,0 +1,679 @@
+
+        args : dict,optional {None}
+            Not supported
+
+        e_ops : list {None}
+            List of Qobj,QobjEvo or callable to compute the expectation
+            values. Function[s] must have the signature
+            f(t : float,state : Qobj) -> expect.
+
+        Returns
+        -------
+        results : :class:`qutip.solver.FloquetResult`
+            Results of the evolution. States and/or expect will be saved. You
+            can control the saved data in the options.
+        """
+
+        if args:
+            raise ValueError("FLiMESolver cannot update arguments")
+        if not floquet:
+            state0 = self.floquet_basis.to_floquet_basis(state0, tlist[0])
+
+        _time_start = time()
+        _data0 = self._prepare_state(state0)
+        self._integrator.set_state(tlist[0], _data0)
+        stats = self._initialize_stats()
+        results = self.resultclass(
+            e_ops,
+            self.options,
+            solver=self.name,
+            stats=stats,
+            floquet_basis=self.floquet_basis,
+        )
+
+        if state0.type == "ket":
+            state0 = operator_to_vector(state0 * state0.dag())
+        elif state0.type == "oper":
+            state0 = operator_to_vector(state0)
+        else:
+            raise ValueError("You need to supply a valid ket or operator")
+
+        fstates_table = _floquet_state_table(self.floquet_basis, tlist)
+
+        stats["preparation time"] += time() - _time_start
+
+        sols = [self._restore_state(_data0, copy=False).full()]
+        progress_bar = progress_bars[self.options["progress_bar"]](
+            len(tlist) - 1, **self.options["progress_kwargs"]
+        )
+        for t, state in self._integrator.run(tlist):
+            progress_bar.update()
+            sols.append(self._restore_state(state, copy=False).full())
+        progress_bar.finished()
+        sols = np.array(sols)
+
+        sols_comp_arr = np.einsum(
+            "xij,xjk,xkl->xil",
+            fstates_table,
+            sols,
+            np.transpose(fstates_table.conj(), axes=(0, 2, 1)),
+        )
+
+        sols_comp = [
+            Qobj(
+                _data.Dense(state),
+                dims=[
+                    self.floquet_basis.U(0).dims[0],
+                    self.floquet_basis.U(0).dims[0],
+                ],
+                copy=False,
+            )
+            for state in sols_comp_arr
+        ]
+
+        for idx, state in enumerate(sols_comp):
+            results.add(tlist[idx], state, Qobj(fstates_table[idx]))
+
+        stats["run time"] = time() - _time_start
+        return results
+
+    def steadystate(self):
+        """
+        Uses the time-independent part of the Rate matrix to find the
+            steady state of the system. Note: This only uses the full secular
+            approximation.
+        """
+        monomat = self.RateDic[0].full()
+        val, vec = np.linalg.eig(monomat)
```

Comment:
Qutip wrap numpy/scipy eigen sovler, please do not use it directly, but use `Qobj.eigenstates` or other qutip function. In general, any place `Qobj.full` is used means that we will have to fix it later to allow it to run with other back-ends (jax, cupy, ...).

I would prefer you not adding new features at this point and add them in another PR. Otherwise there is always more to review and pushing the eventual merging back.

Reply:
Sounds good. I'll roll it back.